### PR TITLE
Improve the error message for to small values of MAX_CPU_COUNT

### LINKED ISSUE #3594: Error:  The number of OS threads requested (20) does not match the number of threads to bind (12): HPX(bad_parameter)
## Expected Behavior

Example should run 

## Actual Behavior

```
.[diehlpk@centaur hpx]$ ./bin/fibonacci
terminate called after throwing an instance of 'hpx::detail::exception_with_info<hpx::exception>'
  what():  The number of OS threads requested (20) does not match the number of threads to bind (12): HPX(bad_parameter)
Aborted (core dumped)
```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. Compile HPX on  ppc64le
  1. jemalloc-5.0.1
  1. hwloc-1.11.7/
  1. clang 7.0 
  1. boost_1_65_0

## Specifications

... Please describe your environment

  - HPX Version:
```
git clone https://github.com/STEllAR-GROUP/hpx.git
    cd hpx
    git checkout 1.2.0
```

- OS version
 
 ```
[diehlpk@centaur hpx]$ uname -a
Linux centaur 3.10.0-862.9.1.el7.ppc64le #1 SMP Wed Jun 27 08:33:42 UTC 2018 ppc64le ppc64le ppc64le GNU/Linux
```

### PR DESCRIPTION
**Pull request by diehlpk**
Fixes #3594 and #3482

## Proposed Changes

  - Setting a wrong value to `-DHPX_WITH_MAX_CPU_COUNT` results in an unspecific error message

```
[diehlpk@centaur build]$ ./bin/fibonacci
terminate called after throwing an instance of 'hpx::detail::exception_with_info<hpx::exception>'
  what():  The number of OS threads requested (20) does not match the number of threads to bind (12): HPX(bad_parameter)
```
This is misleading and we should improve the error handling here.

## Any background context you want to provide?

The PR shows this error message instead 

```
[diehlpk@centaur build]$ ./bin/fibonacci
terminate called after throwing an instance of 'std::runtime_error'
  what():  partitioner::partioner: HPX_HAVE_MAX_CPU_COUNT is set as 64 your system has 160. Please recompile HPX and increase the maximal CPU count!
Aborted (core dumped)
```

and no error message when the value is correct

```
[diehlpk@centaur build]$ ./bin/fibonacci
fibonacci(10) == 55
elapsed time: 0.002081 [s]
```

### CODE REVIEW

File: src/runtime/resource/detail/detail_partitioner.cpp [hunk 1]
```diff
@@ -225,6 +225,18 @@ namespace hpx { namespace resource { namespace detail
...
+        if(HPX_HAVE_MAX_CPU_COUNT < topo_.get_number_of_pus())
+	{
+                
+		throw_runtime_error(
+                "partitioner::partioner",
+		"HPX_HAVE_MAX_CPU_COUNT is set as " +
+                std::to_string(HPX_HAVE_MAX_CPU_COUNT) +
+                " your system has " +
+                std::to_string(topo_.get_number_of_pus()) +
+                ". Please recompile HPX and increase the maximal CPU count!");
```
Comment (sithhell) [inline src/runtime/resource/detail/detail_partitioner.cpp]: This should point to the right cmake variable or better yet, doc section.

Comment (hkaiser) [reply src/runtime/resource/detail/detail_partitioner.cpp]: I'd suggest to change the error message to:
```
Currently, HPX_HAVE_MAX_CPU_COUNT is set to 64 while your system has 160 
processing units. Please reconfigure HPX with -DHPX_WITH_MAX_CPU_COUNT=160 
(or higher) to increase the maximal CPU count.
```

**Comment by hkaiser** [comment]
@diehlpk Please also fix the error reported by inspect: https://52910-4455628-gh.circle-artifacts.com/0/hpx_inspect_report.html (please replace tabs with spaces and remove all endline whitespace, for our coding style guidelines please see: https://github.com/STEllAR-GROUP/hpx/wiki/HPX-Source-Code-Structure-and-Coding-Standards).

File: src/runtime/resource/detail/detail_partitioner.cpp (hunk 1 above)
Comment (diehlpk) [reply src/runtime/resource/detail/detail_partitioner.cpp]: Sure, I will update the error message

File: src/runtime/resource/detail/detail_partitioner.cpp [hunk 2]
```diff
@@ -226,6 +226,22 @@ namespace hpx { namespace resource { namespace detail
                 "Cannot instantiate more than one resource partitioner");
         }
 
+
+        if(HPX_HAVE_MAX_CPU_COUNT < topo_.get_number_of_pus())
+        {
+        throw_runtime_error(
```
Comment (hkaiser) [inline src/runtime/resource/detail/detail_partitioner.cpp]: The indentation seems to be off here.

**Comment by hkaiser** [comment]
Could you please rebase your changes on master? That should get rid of all the unrelated commits on this PR.

**Comment by diehlpk** [comment]
@hkaiser I totally screwed up my branch and could not push after the rebase and squash anymore, because my branch was behind master, but git merge master and git rebase master said it was up to date. I generated a new PR #3603.

**Pull request closed without merging**